Tests fail

```
$ npm test

> tmp-promise@1.0.0 test C:\Users\ohnob\repos\tmp-promise
> mocha



  withFile
    1) file don't exists after going out of scope

  withDir
    2) dir don't exists after going out of scope


  0 passing (35ms)
  2 failing

  1) withFile file don't exists after going out of scope:
     AssertionError: Missing expected exception..
      at _throws (assert.js:341:5)
      at Function.assert.throws (assert.js:365:3)
      at test.js:23:14
      at tryCatcher (node_modules\bluebird\js\release\util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules\bluebird\js\release\promise.js:510:31)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:567:18)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:580:21)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at Promise._resolveCallback (node_modules\bluebird\js\release\promise.js:431:57)
      at Promise._settlePromiseFromHandler (node_modules\bluebird\js\release\promise.js:522:17)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:567:18)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at Promise._resolveCallback (node_modules\bluebird\js\release\promise.js:431:57)
      at Promise._settlePromiseFromHandler (node_modules\bluebird\js\release\promise.js:522:17)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:567:18)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at PromiseArray._resolve (node_modules\bluebird\js\release\promise_array.js:125:19)
      at PromiseArray._promiseFulfilled (node_modules\bluebird\js\release\promise_array.js:143:14)
      at PromiseArray._iterate (node_modules\bluebird\js\release\promise_array.js:113:31)
      at PromiseArray.init [as _init] (node_modules\bluebird\js\release\promise_array.js:77:10)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:564:21)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at node_modules\bluebird\js\release\nodeback.js:45:21
      at _fileCreated (node_modules\tmp\lib\tmp.js:224:7)
      at FSReqWrap.oncomplete (fs.js:123:15)

  2) withDir dir don't exists after going out of scope:
     AssertionError: Missing expected exception..
      at _throws (assert.js:341:5)
      at Function.assert.throws (assert.js:365:3)
      at test.js:47:14
      at tryCatcher (node_modules\bluebird\js\release\util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules\bluebird\js\release\promise.js:510:31)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:567:18)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:580:21)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at Promise._resolveCallback (node_modules\bluebird\js\release\promise.js:431:57)
      at Promise._settlePromiseFromHandler (node_modules\bluebird\js\release\promise.js:522:17)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:567:18)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at Promise._resolveCallback (node_modules\bluebird\js\release\promise.js:431:57)
      at Promise._settlePromiseFromHandler (node_modules\bluebird\js\release\promise.js:522:17)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:567:18)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at PromiseArray._resolve (node_modules\bluebird\js\release\promise_array.js:125:19)
      at PromiseArray._promiseFulfilled (node_modules\bluebird\js\release\promise_array.js:143:14)
      at PromiseArray._iterate (node_modules\bluebird\js\release\promise_array.js:113:31)
      at PromiseArray.init [as _init] (node_modules\bluebird\js\release\promise_array.js:77:10)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:564:21)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:612:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:691:18)
      at Promise._fulfill (node_modules\bluebird\js\release\promise.js:636:18)
      at node_modules\bluebird\js\release\nodeback.js:45:21
      at _dirCreated (node_modules\tmp\lib\tmp.js:311:7)
      at FSReqWrap.oncomplete (fs.js:123:15)



npm ERR! Test failed.  See above for more details.

ohnob@DESKTOP-A4G2C0J MSYS ~/repos/tmp-promise
$
```

With #10, I can verify that the file and folder created in the tests by `withFile()` and `withDir()` persist even after the test suite exits. E.g.:

```
  1) withFile file doesn't exist after going out of scope:
     AssertionError: Missing expected exception. C:\Users\ohnob\AppData\Local\Temp\tmp-15096dhzvbdjiiDeC.txt still exists
```

and I find during examination that the file still exists.

```
  2) withDir dir doesn't exist after going out of scope:
     AssertionError: Missing expected exception. C:\Users\ohnob\AppData\Local\Temp\tmp-15096sCfkI4CwSHsk.dir still exists
```

And after examining I find that the directory still exists.

--- a/index.js
+++ b/index.js
@@ -21,7 +21,12 @@ module.exports.withFile = function withFile(fn) {
     cleanup = o.cleanup;
     delete o.cleanup;
     return fn(o);
-  }).finally(cleanup);
+  }).finally(function () {
+    // May not pass any arguments to cleanup() or it fails.
+    if (cleanup) {
+      cleanup();
+    }
+  });
 };
 
 
@@ -44,7 +49,12 @@ module.exports.withDir = function withDir(fn) {
     cleanup = o.cleanup;
     delete o.cleanup;
     return fn(o);
-  }).finally(cleanup);
+  }).finally(function () {
+    // May not pass any arguments to cleanup() or it fails.
+    if (cleanup) {
+      cleanup();
+    }
+  });
 };
 
 

